packages/react-aria/src/hooks/useAriaButton.ts

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { ObjectShorthandProps, useEventCallback } from '@fluentui/react-utilities';
+
+export function useARIAButton(shorthand: ObjectShorthandProps<React.ButtonHTMLAttributes<HTMLElement>>) {
+  if (!shorthand.hasOwnProperty('as')) {
+    shorthand.as = 'button';
+  }
+  if (!shorthand.hasOwnProperty('children')) {
+    shorthand.children = null;
+  }
+
+  const { onClick, onKeyDown, onKeyUp, as, disabled } = shorthand;
+
+  const onClickHandler = useEventCallback((ev: React.MouseEvent<HTMLElement>) => {
+    if (disabled) {
+      ev.preventDefault();
+    } else {
+      onClick?.(ev);
+    }
+  });
+
+  const onKeyDownHandler = useEventCallback((ev: React.KeyboardEvent<HTMLElement>) => {
+    onKeyDown?.(ev);
+    if (ev.isDefaultPrevented()) {
+      return;
+    }
+    if (disabled && (ev.key === 'Enter' || ev.key === ' ')) {
+      ev.preventDefault();
+      ev.stopPropagation();
+      return;
+    }
+    if (ev.key === ' ') {
+      ev.preventDefault();
+    }
+    // If enter is pressed, activate the button
+    else if (ev.key === 'Enter') {
+      ev.preventDefault();
+      ev.currentTarget.click();
+    }
+  });
+
+  const onKeyupHandler = useEventCallback((ev: React.KeyboardEvent<HTMLElement>) => {
+    onKeyUp?.(ev);
+    if (ev.isDefaultPrevented()) {
+      return;
+    }
+    if (disabled && (ev.key === 'Enter' || ev.key === ' ')) {
+      ev.preventDefault();
+      ev.stopPropagation();
+      return;
+    }
+    if (ev.key === ' ') {
+      ev.preventDefault();
+      ev.currentTarget.click();
```

Comment:
by using `currentTarget` if there is a focusable inside the `button` then you actually end up triggering click for the inner element at the same time triggering click for the `button`. Is this intentional ?

Reply:
https://codesandbox.io/s/brave-bose-yhvpo